src/Utils.gs

```
@@ -103,9 +130,46 @@ function dateConversionRequired (str) {
   var substrings = ['year', 'month', 'week', ' day', 'hour', 'minute', 'second']
   for (var i = 0; i != substrings.length; i++) {
     var substring = substrings[i]
+
     if (str.indexOf(substring) != - 1) {
       return true
     }
   }
   return false
 }
+
+function logScheduledMessage(message){
+    var row = [message.getId(), message.getTo(), message.getSubject(), message.getDate().toString(), 'Scheduled']
+    var spreadsheet = getLogSpreadsheet()
+    var sheet = spreadsheet.getSheets()[0]
+    sheet.appendRow(row)
+}
+
+function colorHeaderLogFile(sheet){
+    var range = sheet.getRange(1,1,1,5)
+    range.setBackground('#B7D6DD')
+    range.setFontWeight('bold')
+}
+
+function logMessageAsSent(messageId){
+    var spreadsheet = getLogSpreadsheet()
+    var sheet = spreadsheet.getSheets()[0]
+    var range = sheet.getDataRange()
+    var row = getRowByMessageId(messageId, range)
+
+    if (row > 0){
+        sheet.getRange(row, 5).setValue('Sent')
+    } else {
+        Logger.log('Message ' + messageId + 'not found')
```

Let this raise an exception